integration/entity.go

```diff
@@ -0,0 +1,91 @@
+package integration
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/newrelic/infra-integrations-sdk/metric"
+	"github.com/newrelic/infra-integrations-sdk/persist"
+	"github.com/pkg/errors"
+)
+
+// Entity is the producer of the data. Entity could be a host, a container, a pod, or whatever unit of meaning.
+type Entity struct {
+	lock      sync.Mutex
+	Metadata  EntityMetadata   `json:"entity"`
+	Metrics   []*metric.Set    `json:"metrics"`
```

Comment:
At the end... we decided to leave this as a pointer?

Reply:
We need it like that as we are allowing the sdk user to manage the returned entity without updating the integration.

Reply:
```go
type Set struct {
  	storer  persist.Storer
  	Metrics map[string]interface{}
}
```
> we are allowing the sdk user to manage the returned entity

The user will be allowed to change the metrics inside the metric set no matters if `metric.Set` Metrics field is slice to pointer or not.

Reply:
When returned in a struct maps are copied by value, I mean not the pointer value is copied.